addons/common/functions/fnc_addExplosionEventHandler.sqf

```diff
@@ -0,0 +1,36 @@
+#include "..\script_component.hpp"
+/*
+ * Author: johnb432, PabstMirror
+ * Adds event hander for explosions (for local projectiles)
+ *
+ * Arguments:
+ * 0: Code to execute <CODE>
+ *
+ * Return Value:
+ * The CBA Event Handler Index <NUMBER>
+ *
+ * Example:
+ * [{systemChat "boom"}] call ace_common_fnc_addExplosionEventHandler
+ *
+ * Public: Yes
+ */
+
+params [["_code", {}, [{}]]];
+
+if !(missionNamespace getVariable [QGVAR(addExplosionEventHandlerActive), false]) then {
+    GVAR(addExplosionEventHandlerActive) = true;
+
+    addMissionEventHandler ["ProjectileCreated", {
+        params ["_projectile"];
+        TRACE_2("pc",typeOf _projectile,local _projectile);
+
+        if (!local _projectile) exitWith {}; // Rockets only explode on local clients
```

Comment:
Grenades return true for `local` and trigger the explode EH to every client in a MP session. An alternate once 2.18 is released is `isReal` as part of `getShotInfo`.